src/main/java/uk/gov/hmcts/reform/preapi/batch/application/services/reporting/LoggingService.java

```diff
@@ -120,21 +154,31 @@ public void setDebugEnabled(boolean enabled) {
     // PROGRESS TRACKING
     // ==============================
 
-    public void incrementProgress() {
-        processedRecords++;
-        refreshProgressBar();
+    public synchronized void startRun(String label, int total) {
+        this.totalRecords = Math.max(total, 0);
+        this.processedRecords = 0; 
+        this.startTime = LocalDateTime.now();
+        logInfo("Found %,d %s to process", totalRecords, label == null ? "items" : label);
     }
 
-    private synchronized void refreshProgressBar() {
-        int progressWidth = 40;
-        double percentage = Math.min((processedRecords * 100.0) / totalRecords, 100.0);
-        int filledLength = Math.max((int) (progressWidth * (percentage / 100)), 0);
+    public synchronized void markHandled() {
+        handled++;
+        if (handled % LOG_EVERY_N == 0 || handled == totalRecords) {
+            double pct = totalRecords > 0 ? (handled * 100.0) / totalRecords : 0.0;
+            logInfo("Handled %,d of %,d (%.1f%%)", handled, totalRecords, pct);
+        }
+    }
 
-        String progressBar = "[" + "=".repeat(filledLength) + " ".repeat(progressWidth - filledLength) + "]";
-        String progressText = String.format("Processing: %d/%d (%.1f%%)", processedRecords, totalRecords, percentage);
+    public synchronized void markSuccess() {
+        processedRecords++;
+        if (processedRecords % LOG_EVERY_N == 0 || processedRecords == totalRecords) {
+            logInfo("PROGRESS - Processed %,d of %,d (%.1f%%)",
+                processedRecords, totalRecords, progressPercentage());
+        }
+    }
 
-        System.out.print("\r\033[K" + progressBar + " " + progressText);
-        System.out.flush();
+    private double progressPercentage() {
+        return totalRecords > 0 ? Math.min((processedRecords * 100.0) / totalRecords, 100.0) : 0.0;
```

Comment:
I think this is the correct formula for doing percentage of processedRecords of totalRecords?
```suggestion
        return totalRecords > 0 ? Math.min((processedRecords / totalRecords) * 100.0, 100.0) : 0.0;
```

Reply:
Hey @jasonpaige, thanks for reviewing! - so we want to multiple the processedRecords by 100 to make sure we're working with doubles (both these two vars are integer - so e.g. if 200/700 will evaluate to 0 since it'll truncate anyhting after the decimal point)
I've tested this with a small local run, and should work